### PR TITLE
Sharing :  Step one -> create Add link button in sharing tab. Also modify the path in Sharin…

### PR DESCRIPTION
**Pull request by TSI-ashishsharma**
Add a new separate section for the files_sharing app and add a new link button. 

We removed the checkboxes 'Allow editing', 'Allow creating' and 'Allow deleting' and add 'advance permission' and 'unshare' for links and for internal users 'send new mail' option.

Link Share:Add new add link button and when we click on this new link share will be created with text 'link to file'
Remove 'others with access' and 'internal link'

### CODE REVIEW

File: magentacloud25/custom/apps/files_sharing/src/components/SharingEntryLink.vue [hunk 1]
```diff
@@ -0,0 +1,858 @@
...
+import NcActionButton from '@nextcloud/vue/dist/Components/NcActionButton'
+import NcActionCheckbox from '@nextcloud/vue/dist/Components/NcActionCheckbox'
+import NcActionInput from '@nextcloud/vue/dist/Components/NcActionInput'
+import NcActionLink from '@nextcloud/vue/dist/Components/NcActionLink'
+import NcActionText from '@nextcloud/vue/dist/Components/NcActionText'
+import NcActionSeparator from '@nextcloud/vue/dist/Components/NcActionSeparator'
+import NcActionTextEditable from '@nextcloud/vue/dist/Components/NcActionTextEditable'
+import NcActions from '@nextcloud/vue/dist/Components/NcActions'
+import NcAvatar from '@nextcloud/vue/dist/Components/NcAvatar'
+import ExternalShareAction from '../../../../../../../release25.0.6/apps/files_sharing/src/components/ExternalShareAction.vue'
```
Comment (TSI-nareshrai) [inline magentacloud25/custom/apps/files_sharing/src/components/SharingEntryLink.vue]: Not valid path ...

Comment (TSI-nareshrai) [reply magentacloud25/custom/apps/files_sharing/src/components/SharingEntryLink.vue]: Need to updated everywhere

**Comment by tsdicloud** [comment]
Ashish, honestly, you really should think thoroughly whether your design split design is actually merging and conforming to new delivery model. Even without merging, conflicts of the sharing PR are more than likely, and I don't won't to hear any tryouts if you have to deliver cleanly first time.

Please consider re-reading the concept and think this over.

**Comment by tsdicloud** [comment]
Well known rule of thumbing is that copies of files that you want to overwrite MUST be in the äquivalent directory, this has not changed since 2021. So `magentacloud25/^custom^/apps` is an invalid try to avoid detection of impact for the overwrite, directors MUST be `magentacloud25/apps` to keep stable relationship to upstream code.

Invalid design, will unmark!

**Pull request closed without merging**